ssms/basic_simulators/simulator.py

```diff
@@ -744,6 +744,86 @@ def simulator(
         )
 
     # Multi-particle models
+    
+    lba_sd = 0.1
+
+    if model == 'lba_3_v1':
+        # === perform checks ===
+        # check if number of drift rates matches number of actions
+        if 3 != theta[:, :3].shape[1]:
+            raise ValueError('Number of actions does not match number of drift rates')
+        
+        # check if drift rates sum to 1
+        v_sum = np.sum(theta[:, :3], axis = 1)
+        if np.any(v_sum <= 0.99) or np.any(v_sum >= 1.01):
+            raise ValueError('Drift rates do not sum to 1 for each trial')
+        
+        # check if z < a for all trials
+        if np.any(theta[:, [4]] >= theta[:, [3]]):
+            raise ValueError('Starting point z >= a for at least one trial')
+    
+        x = cssm.lba_vanilla_wo_ndt(v = theta[:, :3],
+                            a = theta[:, [3]],
+                            z = theta[:, [4]],
+                            sd = lba_sd,
+                            n_samples = n_samples,
+                            n_trials = n_trials,
+                            max_t = 3
+                            )
+    
+    if model == 'lba_angle_3_v1':
+        # === perform checks ===
+        # check if number of drift rates matches number of actions
+        if 3 != theta[:, :3].shape[1]:
+            raise ValueError('Number of actions does not match number of drift rates')
+        
+        # check if drift rates sum to 1
+        v_sum = np.sum(theta[:, :3], axis = 1)
+        if np.any(v_sum <= 0.99) or np.any(v_sum >= 1.01):
+            raise ValueError('Drift rates do not sum to 1 for each trial')
+        
+        # check if z < a for all trials
+        if np.any(theta[:, [4]] >= theta[:, [3]]):
+            raise ValueError('Starting point z >= a for at least one trial')
+        
+        x = cssm.lba_angle_wo_ndt(v = theta[:, :3],
+                            a = theta[:, [3]],
+                            z = theta[:, [4]],
+                            theta = theta[:, [5]],
+                            sd = lba_sd,
+                            n_samples = n_samples,
+                            n_trials = n_trials,
+                            max_t = 3
+                            )
+    
+    if model == 'rlwm_lba_race_wo_ndt_v1':
+        # === perform checks ===
+        # check if number of drift rates matches number of actions
+        if 3 != theta[:, :3].shape[1]:
```

Comment:
similar to comment below when it comes to parameter transformation, we can have a a `parameter_validity_check()` here which usually just checks the hypercube as a default?

This is also towards more general framework, which I hadn't come around to yet. 

Idea is that ultimately would be great to sort of add this to the model config directly, then have a separation of concerns, where one can think separately about a, potentially complicated, function that makes sure some intricate constraints are satisfied about parameter space without having to worry about making changes to `simulator()` for that part.